Error fetching field values for any BigQuery table with an array (REPEATED) field

**Describe the bug**
Error fetching field values for any BigQuery table with an array (REPEATED) field. As a result (I believe?) the schema of this table cannot be updated in Metabase as the crash in getting field values prevents the sync from completing.

The reason the field-values query fails is array fields are not valid in an ORDER BY and/or a GROUP BY statement in BigQuery.

**Logs**
<details><summary>Full stacktrace</summary>

```
[a18fb46d-a65f-4f81-a1f0-f5285b565616] 2021-04-21T22:23:55+01:00 ERROR metabase.models.field-values Error fetching field values
clojure.lang.ExceptionInfo: Error executing query {:type :invalid-query, :sql "SELECT `reports.orders_plus_all`.`segmentIds` AS `segmentIds` FROM `reports.orders_plus_all` GROUP BY `segmentIds` ORDER BY `segmentIds` ASC LIMIT 5000", :parameters nil}
	at metabase.driver.bigquery$execute_bigquery.invokeStatic(bigquery.clj:224)
	at metabase.driver.bigquery$execute_bigquery.invoke(bigquery.clj:203)
	at metabase.driver.bigquery$execute_bigquery.invokeStatic(bigquery.clj:205)
	at metabase.driver.bigquery$execute_bigquery.invoke(bigquery.clj:203)
	at metabase.driver.bigquery$process_native_STAR_$thunk__1596.invoke(bigquery.clj:277)
	at metabase.driver.bigquery$process_native_STAR_.invokeStatic(bigquery.clj:279)
	at metabase.driver.bigquery$process_native_STAR_.invoke(bigquery.clj:272)
	at metabase.driver.bigquery$eval1600$fn__1602.invoke(bigquery.clj:299)
	at clojure.lang.MultiFn.invoke(MultiFn.java:244)
	at metabase.query_processor.context$executef.invokeStatic(context.clj:59)
	at metabase.query_processor.context$executef.invoke(context.clj:48)
	at metabase.query_processor.context.default$default_runf.invokeStatic(default.clj:68)
	at metabase.query_processor.context.default$default_runf.invoke(default.clj:66)
	at metabase.query_processor.context$runf.invokeStatic(context.clj:45)
	at metabase.query_processor.context$runf.invoke(context.clj:39)
	at metabase.query_processor.reducible$pivot.invokeStatic(reducible.clj:34)
	at metabase.query_processor.reducible$pivot.invoke(reducible.clj:31)
	at metabase.query_processor.middleware.mbql_to_native$mbql__GT_native$fn__47231.invoke(mbql_to_native.clj:25)
	at metabase.query_processor.middleware.check_features$check_features$fn__46483.invoke(check_features.clj:41)
	at metabase.query_processor.middleware.limit$limit$fn__47217.invoke(limit.clj:37)
	at metabase.query_processor.middleware.cache$maybe_return_cached_results$fn__45947.invoke(cache.clj:211)
	at metabase.query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__47396.invoke(optimize_datetime_filters.clj:133)
	at metabase.query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__45093.invoke(auto_parse_filter_values.clj:43)
	at metabase.query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__40741.invoke(wrap_value_literals.clj:147)
	at metabase.query_processor.middleware.annotate$add_column_info$fn__40626.invoke(annotate.clj:582)
	at metabase.query_processor.middleware.permissions$check_query_permissions$fn__46358.invoke(permissions.clj:75)
	at metabase.query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__47919.invoke(pre_alias_aggregations.clj:40)
	at metabase.query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46556.invoke(cumulative_aggregations.clj:60)
	at metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__49491.invoke(row_level_restrictions.clj:331)
	at metabase.query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__48232.invoke(resolve_joined_fields.clj:94)
	at metabase.query_processor.middleware.resolve_joins$resolve_joins$fn__48537.invoke(resolve_joins.clj:178)
	at metabase.query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__44687.invoke(add_implicit_joins.clj:181)
	at metabase.query_processor.middleware.large_int_id$convert_id_to_string$fn__47192.invoke(large_int_id.clj:44)
	at metabase.query_processor.middleware.format_rows$format_rows$fn__47172.invoke(format_rows.clj:74)
	at metabase.query_processor.middleware.desugar$desugar$fn__46622.invoke(desugar.clj:21)
	at metabase.query_processor.middleware.binning$update_binning_strategy$fn__45453.invoke(binning.clj:228)
	at metabase.query_processor.middleware.resolve_fields$resolve_fields$fn__46165.invoke(resolve_fields.clj:24)
	at metabase.query_processor.middleware.add_dimension_projections$add_remapping$fn__44317.invoke(add_dimension_projections.clj:316)
	at metabase.query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__44548.invoke(add_implicit_clauses.clj:146)
	at metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__49491.invoke(row_level_restrictions.clj:331)
	at metabase.query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__48967.invoke(upgrade_field_literals.clj:45)
	at metabase.query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__44840.invoke(add_source_metadata.clj:122)
	at metabase_enterprise.sandbox.query_processor.middleware.column_level_perms_check$maybe_apply_column_level_perms_check$fn__49008.invoke(column_level_perms_check.clj:25)
	at metabase.query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__48116.invoke(reconcile_breakout_and_order_by_bucketing.clj:97)
	at metabase.query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45040.invoke(auto_bucket_datetimes.clj:139)
	at metabase.query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46212.invoke(resolve_source_table.clj:45)
	at metabase.query_processor.middleware.parameters$substitute_parameters$fn__47901.invoke(parameters.clj:111)
	at metabase.query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46264.invoke(resolve_referenced.clj:79)
	at metabase.query_processor.middleware.expand_macros$expand_macros$fn__46878.invoke(expand_macros.clj:155)
	at metabase.query_processor.middleware.add_timezone_info$add_timezone_info$fn__44849.invoke(add_timezone_info.clj:15)
	at metabase.query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__48903.invoke(splice_params_in_response.clj:32)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48127$fn__48131.invoke(resolve_database_and_driver.clj:31)
	at metabase.driver$do_with_driver.invokeStatic(driver.clj:60)
	at metabase.driver$do_with_driver.invoke(driver.clj:56)
	at metabase.query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48127.invoke(resolve_database_and_driver.clj:25)
	at metabase.query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47118.invoke(fetch_source_query.clj:274)
	at metabase.query_processor.middleware.store$initialize_store$fn__48912$fn__48913.invoke(store.clj:11)
	at metabase.query_processor.store$do_with_store.invokeStatic(store.clj:44)
	at metabase.query_processor.store$do_with_store.invoke(store.clj:38)
	at metabase.query_processor.middleware.store$initialize_store$fn__48912.invoke(store.clj:10)
	at metabase.query_processor.middleware.validate$validate_query$fn__48974.invoke(validate.clj:10)
	at metabase.query_processor.middleware.normalize_query$normalize$fn__47244.invoke(normalize_query.clj:22)
	at metabase.query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__44705.invoke(add_rows_truncated.clj:35)
	at metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31311.invoke(handle_audit_queries.clj:162)
	at metabase.query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__48888.invoke(results_metadata.clj:147)
	at metabase.query_processor.reducible$async_qp$qp_STAR___33119$thunk__33120.invoke(reducible.clj:103)
	at metabase.query_processor.reducible$async_qp$qp_STAR___33119.invoke(reducible.clj:109)
	at metabase.query_processor.reducible$async_qp$qp_STAR___33119.invoke(reducible.clj:96)
	at metabase.query_processor.reducible$sync_qp$qp_STAR___33128$fn__33129.invoke(reducible.clj:131)
	at metabase.query_processor.reducible$sync_qp$qp_STAR___33128.invoke(reducible.clj:130)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.query_processor$process_query.invokeStatic(query_processor.clj:146)
	at metabase.query_processor$process_query.doInvoke(query_processor.clj:140)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at metabase.db.metadata_queries$qp_query$fn__51985.invoke(metadata_queries.clj:21)
	at metabase.db.metadata_queries$qp_query.invokeStatic(metadata_queries.clj:20)
	at metabase.db.metadata_queries$qp_query.invoke(metadata_queries.clj:18)
	at metabase.db.metadata_queries$field_query.invokeStatic(metadata_queries.clj:30)
	at metabase.db.metadata_queries$field_query.invoke(metadata_queries.clj:28)
	at metabase.db.metadata_queries$fn__52005$field_distinct_values__52014$fn__52017.invoke(metadata_queries.clj:75)
	at metabase.db.metadata_queries$fn__52005$field_distinct_values__52014.invoke(metadata_queries.clj:68)
	at metabase.db.metadata_queries$fn__52005$field_distinct_values__52014$fn__52015.invoke(metadata_queries.clj:72)
	at metabase.db.metadata_queries$fn__52005$field_distinct_values__52014.invoke(metadata_queries.clj:68)
	at clojure.lang.Var.invoke(Var.java:384)
	at metabase.models.field_values$distinct_values.invokeStatic(field_values.clj:123)
	at metabase.models.field_values$distinct_values.invoke(field_values.clj:115)
	at metabase.models.field_values$create_or_update_field_values_BANG_.invokeStatic(field_values.clj:145)
	at metabase.models.field_values$create_or_update_field_values_BANG_.doInvoke(field_values.clj:139)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at metabase.models.field_values$create_field_values_if_needed_BANG_.invokeStatic(field_values.clj:210)
	at metabase.models.field_values$create_field_values_if_needed_BANG_.doInvoke(field_values.clj:202)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at metabase.api.field$field__GT_values.invokeStatic(field.clj:187)
	at metabase.api.field$field__GT_values.invoke(field.clj:181)
	at metabase.api.field$check_perms_and_return_field_values.invokeStatic(field.clj:197)
	at metabase.api.field$check_perms_and_return_field_values.invoke(field.clj:193)
	at metabase_enterprise.sandbox.api.field$fn__54982.invokeStatic(field.clj:37)
	at metabase_enterprise.sandbox.api.field$fn__54982.invoke(field.clj:30)
	at compojure.core$wrap_response$fn__11947.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__11931.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__11936.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__11940.invoke(core.clj:151)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__55043.invoke(auth.clj:14)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at compojure.core$make_context$handler__11987.invoke(core.clj:287)
	at compojure.core$make_context$fn__11989.invoke(core.clj:296)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$make_context$fn__11989.invoke(core.clj:297)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at metabase.server.routes$fn__80976$fn__80977.doInvoke(routes.clj:57)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at compojure.core$make_context$handler__11987.invoke(core.clj:287)
	at compojure.core$make_context$fn__11989.invoke(core.clj:296)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__11940.invoke(core.clj:152)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__11940.invoke(core.clj:152)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__11940.invoke(core.clj:152)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:199)
	at compojure.core$routes$fn__11959$f__11960$respond_SINGLEQUOTE___11961.invoke(core.clj:197)
	at compojure.core$make_context$fn__11989.invoke(core.clj:297)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at compojure.core$routes$fn__11959$f__11960.invoke(core.clj:198)
	at compojure.core$routes$fn__11959.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__79054.invoke(exceptions.clj:98)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__79051.invoke(exceptions.clj:86)
	at metabase.server.middleware.log$log_api_call$fn__80892$fn__80893.invoke(log.clj:195)
	at toucan.db$_do_with_call_counting.invokeStatic(db.clj:216)
	at toucan.db$_do_with_call_counting.invoke(db.clj:209)
	at metabase.server.middleware.log$log_api_call$fn__80892.invoke(log.clj:189)
	at metabase.server.middleware.security$add_security_headers$fn__79016.invoke(security.clj:142)
	at metabase.server.middleware.json$wrap_json_body$fn__80673.invoke(json.clj:62)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__80691.invoke(json.clj:98)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__81233.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__81249.invoke(params.clj:69)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__35107.invoke(misc.clj:58)
	at metabase.server.middleware.session$bind_current_user$fn__42539$fn__42540.invoke(session.clj:277)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:258)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:250)
	at metabase.server.middleware.session$bind_current_user$fn__42539.invoke(session.clj:276)
	at metabase.server.middleware.session$wrap_current_user_info$fn__42526.invoke(session.clj:236)
	at metabase.server.middleware.session$wrap_session_id$fn__42512.invoke(session.clj:182)
	at metabase.server.middleware.auth$wrap_api_key$fn__55051.invoke(auth.clj:27)
	at ring.middleware.cookies$wrap_cookies$fn__81153.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__35090.invoke(misc.clj:27)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__35115.invoke(misc.clj:75)
	at ring.middleware.gzip$wrap_gzip$fn__81195.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__35118.invoke(misc.clj:92)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__80909.invoke(ssl.clj:48)
	at metabase.server$async_proxy_handler$fn__80506.invoke(server.clj:71)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:556)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: clojure.lang.ExceptionInfo: Grouping by expressions of type ARRAY is not allowed at [1:103] {"code" 400, "errors" [{"domain" "global", "location" "q", "locationType" "parameter", "message" "Grouping by expressions of type ARRAY is not allowed at [1:103]", "reason" "invalidQuery"}], "message" "Grouping by expressions of type ARRAY is not allowed at [1:103]", "status" "INVALID_ARGUMENT"}
	at metabase.driver.google$execute_no_auto_retry.invokeStatic(google.clj:41)
	at metabase.driver.google$execute_no_auto_retry.invoke(google.clj:33)
	at metabase.driver.google$execute.invokeStatic(google.clj:55)
	at metabase.driver.google$execute.invoke(google.clj:45)
	at metabase.driver.bigquery$execute_bigquery.invokeStatic(bigquery.clj:216)
	... 190 more
```

</details>

**To Reproduce**
Steps to reproduce the behavior:
1. Create a view (so that you can easily update the schema in step 3) with a REPEATED field in BigQuery
2. Set up the dataset containing that table as a database in Metabase
3. Go back to BigQuery and remove a field from this view
4. Attempt to use that table in Metabase after re-syncing.
5. See screenshot below (in this case `locale` is the name of the field that was removed)

**Expected behavior**
Ideally I expect Metabase to support array fields in tables.
But at a minimum if I hide an array field I expect Metabase to not go and try and to fetch field values for it and fail to do so.

**Screenshots**
If applicable, add screenshots to help explain your problem.

<img width="1491" alt="Screenshot 2021-04-21 at 22 36 15" src="https://user-images.githubusercontent.com/56902/115624065-fe2b4a00-a2f1-11eb-8c19-1e2d4757b01a.png">


**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-gb",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.1 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "4.14.219-164.354.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "bigquery"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-04-08",
      "tag": "v1.38.4",
      "branch": "release-x.38.x",
      "hash": "057b6bb"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
This prevents using this table even when I disable this array field from being visible anywhere. I cannot remove this field from the original table as it is used needed elsewhere.

**Additional context**
Add any other context about the problem here.

--- a/modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj
+++ b/modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj
@@ -17,8 +17,8 @@
             [schema.core :as s])
   (:import com.google.auth.oauth2.ServiceAccountCredentials
            [com.google.cloud.bigquery BigQuery BigQuery$DatasetOption BigQuery$JobOption BigQuery$TableListOption
-                                      BigQuery$TableOption BigQueryOptions DatasetId EmptyTableResult Field FieldValue
-                                      FieldValueList QueryJobConfiguration Schema Table TableId TableResult]
+                                      BigQuery$TableOption BigQueryOptions DatasetId EmptyTableResult Field Field$Mode
+                                      FieldValue FieldValueList QueryJobConfiguration Schema Table TableId TableResult]
            java.io.ByteArrayInputStream
            java.util.Collections))
 
@@ -90,28 +90,38 @@
      (.getTable client (TableId/of project-id dataset-id table-id) empty-table-options)
      (.getTable client dataset-id table-id empty-table-options))))
 
-(defn- bigquery-type->base-type [field-type]
-  (case field-type
-    "BOOLEAN"    :type/Boolean
-    "FLOAT"      :type/Float
-    "INTEGER"    :type/Integer
-    "RECORD"     :type/Dictionary ; RECORD -> field has a nested schema
-    "STRING"     :type/Text
-    "DATE"       :type/Date
-    "DATETIME"   :type/DateTime
-    "TIMESTAMP"  :type/DateTimeWithLocalTZ
-    "TIME"       :type/Time
-    "NUMERIC"    :type/Decimal
-    "BIGNUMERIC" :type/Decimal
-    :type/*))
+(defn- bigquery-type->base-type
+  "Returns the base type for the given BigQuery field's `field-mode` and `field-type`. In BQ, an ARRAY of INTEGER has
+  \"REPEATED\" as the mode, and \"INTEGER\" as the type name.
+
+  If/when we are able to represent complex types more precisely, we may want to capture that information separately.
+  For now, though, we will check if the `field-mode` is \"REPEATED\" and return our :type/Array for that case, then
+  proceed to check the `field-type` otherwise."
+  [field-mode field-type]
+  (if (= Field$Mode/REPEATED field-mode)
+    :type/Array
+    (case field-type
+      "BOOLEAN"    :type/Boolean
+      "FLOAT"      :type/Float
+      "INTEGER"    :type/Integer
+      "RECORD"     :type/Dictionary ; RECORD -> field has a nested schema
+      "STRING"     :type/Text
+      "DATE"       :type/Date
+      "DATETIME"   :type/DateTime
+      "TIMESTAMP"  :type/DateTimeWithLocalTZ
+      "TIME"       :type/Time
+      "NUMERIC"    :type/Decimal
+      "BIGNUMERIC" :type/Decimal
+      :type/*)))
 
 (s/defn ^:private table-schema->metabase-field-info
   [schema :- Schema]
   (for [[idx ^Field field] (m/indexed (.getFields schema))]
-    (let [type-name (.. field getType name)]
+    (let [type-name (.. field getType name)
+          f-mode    (.getMode field)]
       {:name              (.getName field)
        :database-type     type-name
-       :base-type         (bigquery-type->base-type type-name)
+       :base-type         (bigquery-type->base-type f-mode type-name)
        :database-position idx})))
 
 (defmethod driver/describe-table :bigquery-cloud-sdk
@@ -146,8 +156,11 @@
   com.google.api.services.bigquery.model.GetQueryResultsResponse
   (job-complete? [this] (.getJobComplete ^com.google.api.services.bigquery.model.GetQueryResultsResponse this))
 
+  TableResult
+  (job-complete? [_] true)
+
   EmptyTableResult
-  (job-complete? [this] true))
+  (job-complete? [_] true))
 
 (defn do-with-finished-response
   "Impl for `with-finished-response`."

--- a/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
+++ b/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
@@ -315,3 +315,18 @@
                                          :target [:field (mt/id tbl-nm col-nm)]
                                          :value  [param-v]}]}))
                     first))))))))
+
+(deftest sync-table-with-array-test
+  (testing "Tables with ARRAY (REPEATED) columns can be synced successfully"
+    (do-with-temp-obj "table_array_type_%s"
+      (fn [tbl-nm] ["CREATE TABLE `v3_test_data.%s` AS SELECT 1 AS int_col, GENERATE_ARRAY(1,10) AS array_col"
+                    tbl-nm])
+      (fn [tbl-nm] ["DROP TABLE IF EXISTS `v3_test_data.%s`" tbl-nm])
+      (fn [tbl-nm]
+        (is (= {:schema nil
+                :name   tbl-nm
+                :fields #{{:name "int_col", :database-type "INTEGER", :base-type :type/Integer, :database-position 0}
+                          {:name "array_col", :database-type "INTEGER", :base-type :type/Array, :database-position 1}}}
+              (driver/describe-table :bigquery-cloud-sdk (mt/db) {:name tbl-nm}))
+          "`describe-table` should detect the correct base-type for array type columns")))))
+

